markdown/events/2022/bytesize_nftest.md

```diff
@@ -22,3 +22,81 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[8:12](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=492)
+The next thing that I think is really, really powerful is they already have a `generate` command. So you can generate all of these tests for your pipeline and workflows processes. And also really importantly, we haven't tested the full ability of it's functions. Lastly I think that'll really start to change things, especially as we were starting to change to checking our sample sheet with groovy, that might help us specify those and lock some of that down. Lastly, you just have to run nf-test after writing all of those tests. That will generate all of the boilerplate that you need for basic smoke tests of all the processes, workflows, functions, et cetera.
+
+[9:02](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=542)
+Now a quick little walkthrough of some code. This is just a quick test that I wrote in demultiplex, and I don't think the highlighting's working yet. Oh no, there we go. What's really cool here is, we can start to specify params. So if you've gone through and done this in your workflows and thrown it into GitHub Actions and thrown this into a matrix, you know, you start to feel limited after you get to one or two, or you start to really specify out an entire test and it's just not very familiar. I believe you can parameterize these as well. So you can change your inputs and change various params for your workflow.
+
+[9:50](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=590)
+This is just some boilerplate. It boxes each workflow into its own separate execution directory, I believe. Phil and I have had some interesting findings with that. Again, you can check the size. Now what is the really powerful part here is this snapshot feature. What this does is, it basically pulls in the md5sums for you of all of these files. So we have things like metrics, the run manifest here. We have our fastqs that are zipped up and you can specify all these different paths. This actually isn't a perfect example, because you can just list a directory, I found that really powerful. You can just point it there and then it'll go through all the files. What that looks like over here is that you get this snapshot file.
+
+[10:40](https://www.youtube.com/watch?v=K9B7JRkMpQ4&t=640)
+What this does is it just creates a quick little snapshot with the md5sums of all those files that you specified. And you can do standard out, standard error, et cetera as well on those based on the workflow. But we found that to be a little hairy for right now. As you can see, this is automated. And whenever you rerun your test, it checks for these md5sums and will tell you. The other thing is it's really easy when you know you changed some stuff or you know these are going to change. You just run it with the `-update_snapshots` flag and it updates all the snapshots for you automatically. Then you just commit this file and it will check again against them.
```

Comment:
```suggestion
What this does is it just creates a quick little snapshot with the md5sums of all those files that you specified. And you can do standard out, standard error, et cetera as well on those based on the workflow. But we found that to be a little hairy for right now. As you can see, this is automated. And whenever you rerun your test, it checks for these md5sums and will tell you. The other thing is it's really easy when you know you changed some stuff or you know these are going to change. You just run it with the `--update-snapshot` flag and it updates all the snapshots for you automatically. Then you just commit this file and it will check again against them.
```